beta commit

andrasguseo opened the pull request:
This version already works for TEC settings.
There is a whole buncha cleanup needed, like removing settings etc.
Only need to check the main php file, as there are no settings.
I just would like to get a sense check if it looks OK-good or terribly bad.
For now I want to get this in the repo and I'll keep working on it whenever.

Commit:
beta commit

This version already works for TEC settings.

Commit:
Added reset option

Still need to improve this so when user wants to reset there is a check, like either a popup or user needs to enter a word into a field.

Commit:
Added textdomains and text field for reset

Added and fixed text domains for strings.
Added a text field for the reset to prevent accidental clicks.

Commit:
Added notes for reset option

Commit:
Some cleaning

Adjusted some strings.
Adjusted logic for checking file upload.

Commit:
Further cleanup

Cleaned up unneeded files and code.
Updated pot.

Commit:
Updated version number and such

barryhughes commented on readme.txt:
```diff
@@ -0,0 +1,43 @@
+Donate link: http://m.tri.be/29
+Tags: events, calendar
+Requires at least: 4.5
+Tested up to: 5.1
```
Could keep this fresh and bump to 5.1.1?

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+		 * Is Events Calendar PRO active. If yes, we will add some extra functionality.
+		 *
+		 * @return bool
+         *
```
This and the next line: indentation looks weird (I think because you switched from tabs to spaces: we should always use tabs for the initial indent, though).

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+		 */
+		public function construct() {
+
+		    // @todo Delete. Probably not needed.
```
Yep ... if you aren't doing anything in here you can safely remove the entire method.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+		 * Useful for conditionally-requiring a Tribe plugin, whether to add extra functionality
+		 * or require a certain version but only if it is active.
+		 */
+		public function detect_tec_pro() {
```
Looks to me like this method will never be called ... if that's correct then, to keep things tidy, we should probably remove it (or restore the currently commented out call from `construct()`).

andrasguseo replied:
I left it in as I'm not yet sure how I want to handle settings from our other plugins. They use different entries in the db.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+			// Insert filters and hooks here
+			//add_filter( 'thing_we_are_filtering', [ $this, 'my_custom_function' ] );
+			add_action( 'admin_menu', array( $this, 'tribe_settings_menu' ), 99 );
+            add_action( 'admin_init', array( $this, 'tribe_sie_process_settings_export' ) );
```
1. Initial indent should be tabs.
2. This and other lines in the same block: we can optionally switch to the more concise `[]` notation for our callback arrays, ie `[ $this, 'tribe_settings_menu' ]`.

andrasguseo replied:
Good call.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+		function tribe_sie_settings_page() {
+		    ?>
+			<div class="wrap">
+				<h2><?php _e( 'Settings Import / Export', 'PLUGIN_TEXT_DOMAIN' ); ?></h2>
```
We'll want to correct the text domain before shipping.

barryhughes replied:
(Same note applies elsewhere in this change.)

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+			<div class="wrap">
+				<h2><?php _e( 'Settings Import / Export', 'PLUGIN_TEXT_DOMAIN' ); ?></h2>
+
+                <?php
```
This next block: ideally, let's indent using tabs.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+					<div class="postbox">
+						<h3><span><?php _e( 'Export Settings', 'PLUGIN_TEXT_DOMAIN' ); ?></span></h3>
+						<div class="inside">
+							<p><?php _e( 'Export the setting of The Events Calendar, Event Tickets and add-ons for this site as a .json file. This allows you to easily import the configuration into another site.', 'PLUGIN_TEXT_DOMAIN' ); ?></p>
```
```suggestion
							<p><?php esc_html_e( 'Export the setting of The Events Calendar, Event Tickets and add-ons for this site as a .json file. This allows you to easily import the configuration into another site.', 'PLUGIN_TEXT_DOMAIN' ); ?></p>
```

Looks like a string we could and should escape.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+
+				<div class="metabox-holder">
+					<div class="postbox">
+						<h3><span><?php _e( 'Export Settings', 'PLUGIN_TEXT_DOMAIN' ); ?></span></h3>
```
```suggestion
						<h3><span><?php esc_html__e( 'Export Settings', 'PLUGIN_TEXT_DOMAIN' ); ?></span></h3>
```

Looks like a string we could+should escape ... same note may apply elsewhere, too.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+                $action = 'import_failed';
+            }
+
+			wp_safe_redirect( admin_url( 'edit.php?post_type=tribe_events&page=tribe_import_export&action=' . $action ) ); exit;
```
Moving the `exit();` call to its own line would improve readability :-)

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+			// Retrieve the settings from the file and convert the json object to an array.
+			$settings = json_decode( file_get_contents( $import_file ), true );
+
+            if ( update_option( 'tribe_events_calendar_options', $settings ) ) {
```
1. Before we use `$settings` we should probably do some basic validation (if we could not decode it, for instance, it will be `(bool) false` ... we might also test to ensure it is of type `array`, etc—basically, does it look the way we expect it to look?).
2. Possibly something for a future version ... might it be worth capturing the current settings and storing them in a backup option, so we can offer users the ability to quickly undo/rollback.

andrasguseo replied:
#1 is done.
#2 is a fantastic idea, I love it!
Now I want to get this first version out the door. I still want to further enhance it with including options of our premium plugins, and then I will add this as well.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+
+namespace Tribe\Extensions\Settings_Import_Export;
+
+use Tribe__Autoloader;
```
I _suspect_ this is from the extension template, but in this case we don't need or use autoloading and it can be removed.

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+			}
+
+			// Insert filters and hooks here
+			//add_filter( 'thing_we_are_filtering', [ $this, 'my_custom_function' ] );
```
Remove for tidiness!

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+		 * Register the settings page
+		 */
+		function tribe_settings_menu() {
+			//add_options_page( __( 'Sample Settings Import and Export' ), __( 'Sample Settings Import and Export' ), 'manage_options', 'tribe-settings', 'tribe_settings_page' );
```
Remove for tidiness! 

Where possible, use version control to store old code rather than leaving it in place (but commented out).

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,370 @@
+                        <h3><span><?php _e( 'Delete / Reset Settings', 'PLUGIN_TEXT_DOMAIN' ); ?></span></h3>
+                        <div class="inside">
+                            <p><?php _e( 'Reset the plugin settings.', 'PLUGIN_TEXT_DOMAIN'); ?></p>
+                            <p style="font-weight: bold";><?php _e( 'Please note the following:', '' )?></p>
```
We're probably trying to avoid enqueuing a stylesheet etc with these inline styles which is fine, but perhaps using `<strong>` would be cleaner and better in a case like this. If not, we should fix the trailing semi-colon:

```suggestion
                            <p style="font-weight: bold"><?php _e( 'Please note the following:', '' )?></p>
```

barryhughes replied:
...You are missing the `PLUGIN_TEXT_DOMAIN` placeholder on this line, which is fine but just don't remember to replace it once you do settle on a text domain!

barryhughes reviewed (changes requested):
Looks like an interesting extension! Did an initial pass and left some comments :-)

Commit:
Fixed indentation

Commit:
Changing arrays

Commit:
Escaping strings

Commit:
Cleanup

Commit:
Fixed text domain

Commit:
Removed requirements

Commit:
Removed ECP detection

Commit:
Further cleanup

Commit:
Added checks befoe importing

Check if decoding is successful and if it is the right format (array).
Fixed an error message.

Commit:
Updated .pot file

Commit:
Updated release date in readme

Commit:
Additional fixes

Fixed a php warning and a set of brackets.

Commit:
Update tribe-ext-settings-import-export.php

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,328 @@
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( 'tribe-ext-settings-import-export', false, basename( dirname( __FILE__ ) ) . '/languages/' );
```
Totally optional, but since we're no longer writing PHP 5.2 compatible code we could simplify this line:
```suggestion
			load_plugin_textdomain( 'tribe-ext-settings-import-export', false, basename( __DIR__ ) . '/languages/' );
```

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,328 @@
+		/**
+		 * Render the settings page
+		 */
+		function tribe_sie_settings_page() {
```
Totally optional, but you could consider dropping the `tribe_sie_*` prefix from this and various other methods here. Remember that we're already within the `Tribe\Extensions\Settings_Import_Export` namespace, so the value of the prefix is far less than if we were writing an un-namespaced global function.

barryhughes replied:
...Oh, one other note that applies here and to a few other methods: our coding standards are to always be explicit about method visibility. So, in this case, you'd want:
```suggestion
		public function tribe_sie_settings_page() {
```
(As with a few other notes, not vital we fix today.)

barryhughes commented on tribe-ext-settings-import-export.php:
```diff
@@ -0,0 +1,328 @@
+			$settings = json_decode( file_get_contents( $import_file ), true );
+
+			if ( false === $settings ) {
+				wp_die( __( 'Sorry, we could  not decode the file.', 'tribe-ext-settings-import-export' ) );
```
Super minor: extra space (don't _have_ to fix it):
```suggestion
				wp_die( __( 'Sorry, we could not decode the file.', 'tribe-ext-settings-import-export' ) );
```

barryhughes reviewed (approved):
Looks good! Left a few more minor notes but none that are critical. This looks like it could be merged, tested and shipped!

Beyond that, some stuff you could _consider_ in a future iteration (again, no need to address now—far better we 'create and share value quickly') :-)

* Instead of multiple forms in the same option screen, we could have a single form with different submit buttons
* We could have different nonces for each possible action
* If we stick with the same nonce for all actions (or even if we don't) we could probably share the logic used to test for a valid submission instead of duplicating it across several handler methods
* The error/notice handling might be cleaner/tidier if it was transformed into a switch block and moved out of the template itself
* Nice work here!

barryhughes committed:
Simplify load textdomain

Co-Authored-By: andrasguseo <andras@tri.be>

barryhughes committed:
Make function public

Remember to do this for all in the future.

Co-Authored-By: andrasguseo <andras@tri.be>

barryhughes committed:
Space fix

Co-Authored-By: andrasguseo <andras@tri.be>
